index.js

```diff
@@ -0,0 +1,192 @@
+  },
+];
+
+const UserGenderArray = (users) => {
+  const usersGroupedByGender = {
+    women: [],
+    men: [],
+  };
+
+  users.forEach((user) => {
+    const { first_name, last_name, ...userUpd } = user;
+    userUpd.fullName = `${first_name} ${last_name}`;
+
+    if (user.gender === "female") {
+      usersGroupedByGender.women.push(userUpd);
+    } else {
+      usersGroupedByGender.men.push(userUpd);
+    }
+  });
+
+  return usersGroupedByGender;
+};
+
+// Task 2
+/* Преобразуйте массив в объект используя функцию reduce */
+
+const videos = [
+  {
+    id: 65432445,
+    title: "The Chamber",
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+  },
+  {
+    id: 70111470,
+    title: "Die Hard",
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+  },
+];
+
+const ArrayToObject = (array) => {
+  return array.reduce((resArr, object) => {
+    resArr[object.id] = object.title;
+    return resArr;
+  }, {});
+};
+
+// Task 3
+/* Выведите массив ids для релизов у которых рейтинг 5.0. */
+
+const newReleases = [
+  {
+    id: 70111470,
+    title: "Die Hard",
+    boxart: "http://cdn-0.nflximg.com/images/2891/DieHard.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+    boxart: "http://cdn-0.nflximg.com/images/2891/BadBoys.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+  {
+    id: 65432445,
+    title: "The Chamber",
+    boxart: "http://cdn-0.nflximg.com/images/2891/TheChamber.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+const bestReleases = (releasesArray) => {
+  const selectedReleases = [];
+  releasesArray.forEach((obj) =>
+    obj.rating[0] === 5.0 ? selectedReleases.push(obj.id) : null
+  );
+  return selectedReleases;
+};
+
+// Task 4
+/* С помощью функций map, reduce, вывести url у которого самая большая площадь */
+
+const boxarts = [
+  {
+    width: 200,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture200.jpg",
+  },
+  {
+    width: 150,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture150.jpg",
+  },
+  {
+    width: 300,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture300.jpg",
+  },
+  {
+    width: 425,
+    height: 150,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture425.jpg",
+  },
+];
+
+const getBiggestArea = (boxArray) => {
+  return boxArray.reduce((biggestObj, obj) => {
+    biggestObj =
+      biggestObj.width * biggestObj.height > obj.width * obj.height
```

Comment:
suggestion: try to avoid so long statements. don't afraid the creation of variables :)

Reply:
fixed